File: swift/sorting/insertionSort.swift
func insertionSort(_ arr: [Int]) -> [Int] {
  // Swift treats parameters as constants
  var sortArray = arr
  
  // Traverse through 1 to count(arr) - 1
  for i in 0...sortArray.count - 1 {
    var j = i - 1
    while j >= 0 && sortArray[j + 1] < sortArray[j] {
      // arr[j] and arr[j + 1] are out of order so swap them
      let temp = sortArray[j + 1]
      sortArray[j + 1] = sortArray[j]
      sortArray[j] = temp
      j -= 1
    }
  }
  return sortArray
}

File: csharp/sorting/InsertionSort.cs
public class InsertionSort
{
    public static int[] Sort(int[] arr)
    {
        for (int i = 1; i < arr.Length; i++)
        {
            int j = i - 1;
            while (j >= 0 && arr[j + 1] < arr[j])
            {
                int tmp = arr[j + 1];
                arr[j + 1] = arr[j];
                arr[j] = tmp;
                j--;
            }
        }
        return arr;
    }
}
